### PR TITLE
Add unit test to verify that Start() never fails for peer IP gathering issues

### PR DESCRIPTION
**Pull request by slintes**
#### Why we need this PR

A bug was reported that for SNR 0.10.x the operator start fails when populating peer addresses fails.
That bug was introduced by backporting  https://github.com/medik8s/self-node-remediation/pull/259/commits/0d12031ad31489c91cb9a7906709417373645e5a without backporting https://github.com/medik8s/self-node-remediation/commit/c15b385f227faa49910a3dbcce02d3d0bcf40e34

#### Changes made
Verify that Start() indeed never fails because of peer addresses population in various scenarios.

- Covers worker and control plane peers
- Includes cases for missing pods, empty IPs, and mixed node types

#### Which issue(s) this PR fixes
Verifies that [RHWA-382](https://issues.redhat.com//browse/RHWA-382) isn't an issue on main branch / for release 0.11

The test fails as expected on the 0.10 release branch:
```
=== RUN   TestStartVerifiesPeerAddresses/Control_plane_node_with_missing_pod_and_empty_IP
    peers_test.go:626: Start() returned an error when it should not: empty IP for Pod snr-pod-cp-2 on Node cp-2
        Node cp-2 has no matching Pod
        Node cp-3 has no matching Pod
    peers_test.go:680: Test passed: Only cp-1 should be included (cp-2 has empty IP, cp-3 missing pod)
--- FAIL: TestStartVerifiesPeerAddresses/Control_plane_node_with_missing_pod_and_empty_IP (0.10s)
```

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for peer address verification and aggregation functionality. Introduced comprehensive integration-style test scenarios covering edge cases including scenarios with missing pods, empty IP addresses, and mixed deployment node roles. Validates peer address verification logic and error handling expectations across multiple configurations to ensure system reliability and robustness.

### CODE REVIEW

File: pkg/peers/peers_test.go [hunk 1]
```diff
@@ -244,3 +251,433 @@ func TestMapNodesToPrimaryPodIPs(t *testing.T) {
...
+			defer cancel()
+
+			err := p.Start(ctx)
+			if err != nil {
+				t.Errorf("Start() returned an error when it should not: %v", err)
+			}
+
+			// Give it a moment to populate the addresses
+			time.Sleep(100 * time.Millisecond)
+
```
Comment (clobrano) [inline pkg/peers/peers_test.go]: It might be overkill, but the `require` package (https://pkg.go.dev/github.com/stretchr/testify/require) offers the `Eventually` assertion, much like in Ginkgo/Gomega, which might be useful here in place of `sleep`.

**Review by clobrano — APPROVED** [review]
I left a comment, but it is a very minor observation

/lgtm

Leaving time to get more reviews

/hold
